sled-agent/src/services.rs

```diff
@@ -238,6 +283,11 @@ pub struct ServiceManager {
     advertised_prefixes: Mutex<HashSet<Ipv6Subnet<SLED_PREFIX>>>,
 }
 
+#[derive(Clone)]
+pub struct ServiceManager {
+    inner: Arc<ServiceManagerInner>,
+}
```

Comment:
I split this out so background tasks could operate on the `ServiceManager` - that's basically what we're doing with the switch zone. When we detect the driver, we keep on trying to load it, but the *responsibility* of loading it exists in the `ServiceManager`.